aws-core/src/main/java/software/amazon/awssdk/awscore/interceptor/GlobalServiceExecutionInterceptor.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.awscore.interceptor;
+
+import java.net.UnknownHostException;
+import software.amazon.awssdk.auth.AwsExecutionAttributes;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.core.interceptor.Context;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.interceptor.ExecutionInterceptor;
+
+/**
+ * An interceptor that can be used for global services that will tell the customer when they're using a global service that
+ * doesn't support non-global regions.
+ */
+public class GlobalServiceExecutionInterceptor implements ExecutionInterceptor {
```

Comment:
I think forward compatibility is less of a concern in this case. It seems reasonable to expect that customers can upgrade to take advantage of the new regional endpoints should a global service start supporting them.

Reply:
We can talk about it offline a bit. Failing fast would definitely be nice. 

We're usually forward-compatible with new regions on our clients, though. This would be against that, even though it's a little bit different (global to regional), it still may be surprising.